lte/gateway/c/session_manager/LocalEnforcer.cpp

```diff
@@ -262,14 +263,22 @@ void LocalEnforcer::handle_pipelined_response(
   }
 }
 
-void LocalEnforcer::poll_stats_enforcer(int cookie, int cookie_mask) {
+void LocalEnforcer::poll_stats_enforcer(std::vector<int> shard_ids) {
   // we need to pass in a function pointer. Binding is required because
   // the function is part of the LocalEnforcer class and has arguments
   // so we bind to the object and the two arguments the function needs
   // which are the status and RuleRecordTable response
-  pipelined_client_->poll_stats(
-      cookie, cookie_mask,
-      std::bind(&LocalEnforcer::handle_pipelined_response, this, _1, _2));
+  if (shard_ids.empty()) {
```

Comment:
I think we chatted about this already, lets move this logic into StatsPoller, and just call poll_stats_enforcer with different cookie/cookie_mask args

Reply:
Right, I'll move it back another level.